drivers/touch/iqs5xx/iqs5xx.c

```diff
@@ -0,0 +1,343 @@
+			LOG_INF("Swipe Y negative");
+			break;
+		case IQS5XX_SWIPE_Y_POS:
+			LOG_INF("Swipe Y positive");
+			break;
+		default:
+			LOG_INF("%d", data->regmap.gesture_event[0]);
+		}
+	}
+
+	if (data->regmap.gesture_event[1]) {
+		switch (data->regmap.gesture_event[1] & 0b00000111) {
+		case IQS5XX_TWO_FINGER_TAP:
+			LOG_INF("Two Finger Tap");
+			break;
+		case IQS5XX_SCROLL:
+			LOG_INF("Scroll");
+			break;
+		case IQS5XX_ZOOM:
+			LOG_INF("Zoom");
+			break;
+		default:
+			LOG_INF("%d", data->regmap.gesture_event[1]);
+		}
+	}
+#endif
+
+	data->input_serviced = false;
+	setup_int(data->dev, true);
+}
+
+int iqs5xx_init_interrupt(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+	const struct iqs5xx_dev_config *cfg = dev->config;
+	int rc;
+
+	if (!device_is_ready(cfg->rdy_gpio.port)) {
+		LOG_ERR("%s: device %s is not ready", dev->name, cfg->rdy_gpio.port->name);
+		return -ENODEV;
+	}
+
+	rc = gpio_pin_configure_dt(&cfg->rdy_gpio, GPIO_INPUT | cfg->rdy_gpio.dt_flags);
+	if (rc < 0) {
+		return rc;
+	}
+
+	gpio_init_callback(&data->gpio_cb, iqs5xx_gpio_callback, BIT(cfg->rdy_gpio.pin));
+
+	rc = gpio_add_callback(cfg->rdy_gpio.port, &data->gpio_cb);
+	if (rc < 0) {
+		return rc;
+	}
+
+	data->dev = dev;
+
+	data->work.handler = iqs5xx_work_cb;
+	return 0;
+}
+
+static int iqs5xx_two_finger_tap(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	if (data->regmap.gesture_event[0] & IQS5XX_TWO_FINGER_TAP) {
+		data->regmap.gesture_event[0] &= ~IQS5XX_TWO_FINGER_TAP;
+		return true;
+	}
+
+	return false;
+}
+
+static int iqs5xx_single_tap(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	if (data->regmap.gesture_event[0] & IQS5XX_SINGLE_TAP) {
+		data->regmap.gesture_event[0] &= ~IQS5XX_SINGLE_TAP;
+		return true;
+	}
+
+	return false;
+}
+
+static int16_t iqs5xx_x_position_abs(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	return sys_be16_to_cpu(data->regmap.touch_data[0].abs_x);
+}
+
+static int16_t iqs5xx_y_position_abs(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	return sys_be16_to_cpu(data->regmap.touch_data[0].abs_y);
+}
+
+static int16_t iqs5xx_x_position_rel(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	return sys_be16_to_cpu(data->regmap.rel_x);
+}
+
+static int16_t iqs5xx_y_position_rel(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	return sys_be16_to_cpu(data->regmap.rel_y);
+}
+
+static int iqs5xx_num_fingers(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+
+	return data->regmap.num_fin;
+}
+
+static const struct touch_driver_api iqs5xx_driver_api = {
+	.single_tap = &iqs5xx_single_tap,
+	.two_finger_tap = &iqs5xx_two_finger_tap,
+	.x_pos_abs = &iqs5xx_x_position_abs,
+	.y_pos_abs = &iqs5xx_y_position_abs,
+	.x_pos_rel = &iqs5xx_x_position_rel,
+	.y_pos_rel = &iqs5xx_y_position_rel,
+	.num_fingers = &iqs5xx_num_fingers,
```

Comment:
Would it be useful to also have some callbacks to user when specific events occur?